index.d.ts

```diff
@@ -2,17 +2,27 @@ import { EventEmitter } from "events";
 import * as fs from "fs";
 import * as http from "http";
 
+declare interface Configstore {
+    get(key: string) : Promise<IFile | undefined> | IFile | undefined;
```

Comment:
Neither am I. I'm not aware of optional return type in TypeScript. I went over the TypeScript documentation and could not find optional return type. Only optional parameters.

The idea is that Configstore can be synchronous or asynchronous and in both cases it can return `undefined` in case when there is no entry for specified key.

Reply:
I think this type may be what you are looking for: https://github.com/sindresorhus/type-fest/blob/main/source/promisable.d.ts.

In the TypeScript rewrite I'll look more closely at creating the most correct types.